clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocol.java

```diff
@@ -178,8 +153,12 @@ public static ByteBuffer serializeSubscriptionV1(PartitionAssignor.Subscription
         return buffer;
     }
 
-    public static ByteBuffer serializeSubscription(PartitionAssignor.Subscription subscription) {
-        switch (subscription.version()) {
+    public static ByteBuffer serializeSubscription(Subscription subscription) {
+        return serializeSubscription(subscription, CONSUMER_PROTOCL_LATEST_VERSION);
+    }
+
+    public static ByteBuffer serializeSubscription(Subscription subscription, short version) {
```

Comment:
Currently you'd only ever want to serialize as the latest version, so in theory we don't need this or even `serializeSubscription/AssignmentV0`-- but maybe in the future we might want to be able to serialize as lower versions?
We can always add back in though, so maybe it's best to just remove for now -- WDYT?

Reply:
I'm fine keeping those unused functions for now, but again I'd leave to your call :)